tilavarauspalvelu/api/frontend_testing_api/serializers.py

```diff
@@ -0,0 +1,104 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from django.contrib.auth import login
+from django.contrib.auth.hashers import make_password
+from rest_framework import serializers
+
+from tilavarauspalvelu.api.frontend_testing_api.helpers import set_current_datetime
+from tilavarauspalvelu.enums import ReservationNotification, UserRoleChoice
+from tilavarauspalvelu.models import Unit
+
+from tests.factories import GeneralRoleFactory, UnitRoleFactory, UserFactory
+
+if TYPE_CHECKING:
+    import datetime
+
+    from tilavarauspalvelu.models import User
+
+
+class TestingBaseSerializer(serializers.Serializer):
+    def validate(self, attrs: dict[str, Any]) -> dict[str, Any]:
+        """Validate that only allowed parameters are passed"""
+        for attr in attrs:
+            if attr not in self.fields:
+                msg = f"Invalid parameter: {attr}"
+                raise serializers.ValidationError(msg)
+
+        return attrs
+
+    def save(self, **kwargs: Any) -> None:
+        """Simplified version of BaseSerializer.save() without the need to return an instance of the created object"""
+        assert hasattr(self, "_errors"), "You must call `.is_valid()` before calling `.save()`."  # noqa: S101
+        assert not self.errors, "You cannot call `.save()` on a serializer with invalid data."  # noqa: S101
+
+        validated_data = {**self.validated_data, **kwargs}
+
+        self.instance = self.create(validated_data)
```

Comment:
Why is this implemented?

Reply:
Initially added this to remove the need to return anything from `.create()` methods, but removed it now since it isn't really needed 😄